pkg/usageStats/stats.go

```diff
@@ -408,27 +427,30 @@ func UpdateMetricsStats(metricsBytesCount uint64, incomingMetrics uint64, orgid
 	atomic.AddUint64(&ustats[orgid].MetricsBytesCount, metricsBytesCount)
 }
 
-func GetQueryStats(orgid uint64) (uint64, float64, uint64) {
+func GetQueryStats(orgid uint64) (uint64, float64, float64, uint64) {
 	if _, ok := QueryStatsMap[orgid]; !ok {
-		return 0, 0, 0
+		return 0, 0, 0, 0
 	}
-	return QueryStatsMap[orgid].QueryCount, QueryStatsMap[orgid].TotalRespTime, QueryStatsMap[orgid].QueriesSinceInstall
+	return QueryStatsMap[orgid].QueryCount, QueryStatsMap[orgid].TotalRespTimeSinceRestart, QueryStatsMap[orgid].TotalRespTimeSinceInstall, QueryStatsMap[orgid].QueriesSinceInstall
 }
 
 func GetCurrentMetricsStats(orgid uint64) (uint64, uint64) {
 	return ustats[orgid].TotalBytesCount, ustats[orgid].TotalMetricsDatapointsCount
 }
 
-func UpdateQueryStats(queryCount uint64, totalRespTime float64, orgid uint64) {
+func UpdateQueryStats(queryCount uint64, respTime float64, orgid uint64) {
 	if _, ok := QueryStatsMap[orgid]; !ok {
 		QueryStatsMap[orgid] = &QueryStats{
-			QueryCount:    0,
-			TotalRespTime: 0,
+			QueryCount:                0,
+			TotalRespTimeSinceRestart: 0,
+			TotalRespTimeSinceInstall: 0,
 		}
 	}
-	atomic.AddUint64(&QueryStatsMap[orgid].QueryCount, queryCount)
-	atomic.AddUint64(&QueryStatsMap[orgid].QueriesSinceInstall, queryCount)
-	QueryStatsMap[orgid].TotalRespTime += totalRespTime
+	qs := QueryStatsMap[orgid]
+	atomic.AddUint64(&qs.QueryCount, queryCount)
+	atomic.AddUint64(&qs.QueriesSinceInstall, queryCount)
+	qs.TotalRespTimeSinceRestart += respTime
+	qs.TotalRespTimeSinceInstall += respTime
```

Comment:
Why do you have to atomically update `qs.QueryCount` but you don't have to atomically update `qs.TotalRespTimeSinceRestart`?

Reply:
We atomically update `qs.QueryCount` because it is a `uint64` and Go provides atomic operations for integer types to ensure thread safety. We were previously doing the same too; we weren't using atomic updates for `qs.TotalRespTime` because Go does not provide atomic operations for floating-point types.

We can probably use a `sync.Mutex` on `QueryStatsMap` to ensure thread safety for updating `qs.TotalRespTimeSinceRestart` and `qs.TotalRespTimeSinceInstall`? What do you think?

Reply:
Yeah I'm not sure if this code will get called by multiple concurrent threads, but the atomic updates make me think it is. If it is, then yes we should use a lock for the floating point numbers. Or another option would be to change the internal structure so they we use small enough units for the floats that we can use ints instead and then convert back to the units they currently are (seconds?) whenever we show them (e.g., send them to frontend)

Reply:
I have updated the code to ensure thread safety for the floating-point numbers by using a `sync.Mutex`